warehouse/forklift/legacy.py

```diff
@@ -362,6 +366,92 @@ def _is_duplicate_file(db_session, filename, hashes):
     return None
 
 
+def _process_attestations(request, metrics, artifact_path: Path):
```

Comment:
Nitpick: rather than pass in `metrics`, get `metrics` via `request.find_service(...)`

(I'll make this change; just flagging.)